src/entities/pair.ts

```diff
@@ -178,15 +178,18 @@ export class Pair {
    *
    * @param inputAmount
    */
-  public getOutputAmount(inputAmount: CurrencyAmount<Token>): [CurrencyAmount<Token>, Pair] {
+  public getOutputAmount(
+    inputAmount: CurrencyAmount<Token>,
+    calculateFotFees: boolean = false
+  ): [CurrencyAmount<Token>, Pair] {
     invariant(this.involvesToken(inputAmount.currency), 'TOKEN')
     if (JSBI.equal(this.reserve0.quotient, ZERO) || JSBI.equal(this.reserve1.quotient, ZERO)) {
       throw new InsufficientReservesError()
     }
     const inputReserve = this.reserveOf(inputAmount.currency)
     const outputReserve = this.reserveOf(inputAmount.currency.equals(this.token0) ? this.token1 : this.token0)
 
-    const percentAfterSellFees = this.derivePercentAfterSellFees(inputAmount)
+    const percentAfterSellFees = calculateFotFees ? this.derivePercentAfterSellFees(inputAmount) : ZERO_PERCENT
```

Comment:
hmm shouldnt this be 100% instead of 0? this refers to the % of `inputAmount` that goes into the v2 calculation right?

Reply:
The function we use to derive returns ZERO_PERCENT when there's no fee:

```
  private derivePercentAfterSellFees(inputAmount: CurrencyAmount<Token>): Percent {
    const sellFeeBips = inputAmount.currency.sellFeeBps
    if (sellFeeBips?.gt(BigNumber.from(0))) {
      return ONE_HUNDRED_PERCENT.subtract(new Percent(JSBI.BigInt(sellFeeBips)).divide(BASIS_POINTS))
    } else {
      return ZERO_PERCENT
    }
  }

  private derivePercentAfterBuyFees(outputAmount: CurrencyAmount<Token>): Percent {
    const buyFeeBps = outputAmount.currency.buyFeeBps
    if (buyFeeBps?.gt(BigNumber.from(0))) {
      return ONE_HUNDRED_PERCENT.subtract(new Percent(JSBI.BigInt(buyFeeBps)).divide(BASIS_POINTS))
    } else {
      return ZERO_PERCENT
    }
  }
```

so I think that's okay?

Reply:
It needs to be 0%. Although I think we can make it simpler by passing the `calculateFotFees` to `derivePercentAfterBuyFees`